lib/config/fileconf.go

```diff
@@ -220,12 +221,37 @@ func MakeSampleFileConfig(flags SampleFlags) (fc *FileConfig, err error) {
 		Method:    types.JoinMethod(joinMethod),
 	}
 
-	if flags.AuthServer != "" {
-		g.AuthServer = flags.AuthServer
-	}
+	if flags.Version == defaults.TeleportConfigVersionV3 {
```

Comment:
oh man the default flag value is V3. Would be nice for users to have a way say "trust me, I really want a v3 config with auth_server: auth.example.com:4025".

Suggestion: remove the default flag value at the CLI parser level, then in `(*dumpFlags).CheckAndSetDefaults` we should:
1. Let the user set a specific version if they want to, and then trust whatever they give. `teleport configure --version v3 --auth_server auth.example.com:4025` should do exactly what it says.
2. If the user did not set a specific version and set `--auth_server`, default to v2 and print the warning.
3. Else default to v3 (this includes the cases where they set `--proxy` or not)